enhancements/20230316-baremetal-container-mcm-support.md

```diff
@@ -0,0 +1,151 @@
+# Bare-metal Cluster Container and Multi-cluster Management Support
+
+Make the bare-metal Harvester support container workloads and multi-clusters management.
+
+## Summary
+
+This proposal aims to improve Harvester's usability and reduce resource consumption by providing container workload management and multi-cluster management through the bare-metal Harvester cluster.
+
+The current [Rancher integration in Harvester](https://docs.harvesterhci.io/v1.1/rancher/rancher-integration) requires deploying Rancher separately, which can be complicated and resource-intensive. By enabling container workloads and multi-cluster management through Harvester's built-in Rancher will reduce overhead and improve product usability.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/2679
+
+
+## Motivation
+
+### Goals
+
+- Provide support for managing bare-metal container workloads.
+- Provide multi-cluster management(MCM) support through the bare-metal Harvester cluster.
+- Reduce the footprint/overhead of the existing Harvester-Rancher integration method.
+- Support authentication and authorization on the bare-metal Harvester cluster.
+
+
+### Non-goals
+
+- Support built-in Rancher manual upgrade without Harvester's new release.
+- Support importing and managing Harvester with multiple Ranchers.
+- Fleet management support for the Harvester cluster(will be considered as a seperated new feature).
+- Automatically force shutdown user containers that block upgrades. Instead, we may consider draining the node, leaving the user to resolve blocking issues manually, such as volume or pod eviction.
+- User container workloads might block upgrade. We may force shutdown user containers? (or drain the node only, user need to resolve blocking cases manually, e.g, volume or pod can’t be evicted)
+- Backup and restore Rancher (see https://ranchermanager.docs.rancher.com/v2.6/pages-for-subheaders/backup-restore-and-disaster-recovery)
+- Container GPU pass-through support (moved to v1.3.0)
+
+## Proposal
+
+- Provide support for managing bare-metal container workloads
+    - When the `rancher-manager-support` setting is enabled, users can view and manage their container workloads management via the explorer UI which will appears in the hamburger menu.
```

Comment:
@yasker @guangbochen This setting is set to `False` by default. Users need to change the setting to `True` on the settings page and reload the page to see the Rancher-integrated view. Do we want the setting to be `True` by default? Or we can present the choices in the installer GUI to let the user choose.

Reply:
@WuJun2016 I believe UI can help to trigger the page reload if it is required for the configuration changes?

Reply:
We shouldn't set the option to `True` by default since it would be a surprise to users don't know about it. Better to present it in the installer and let users know it can be changed afterwards.